kernel/src/hil/date_time.rs

```diff
@@ -0,0 +1,77 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2022.
+
+//! HIL for Date Time interface
+
+use crate::ErrorCode;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DayOfWeek {
+    Sunday,
+    Monday,
+    Tuesday,
+    Wednesday,
+    Thursday,
+    Friday,
+    Saturday,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Month {
+    January,
+    February,
+    March,
+    April,
+    May,
+    June,
+    July,
+    August,
+    September,
+    October,
+    November,
+    December,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub struct Date {
```

Comment:
```suggestion
pub struct DateTime {
```
?

Reply:
The name is supposed to differentiate easily between the DateTime trait (for getting and setting the date) and the actual Date structure getting passed around.

Reply:
Maybe `DateTimeValues`? This struct contains more than just date information.

Reply:
That is reasonable. I will rename the struct.